File: FilmFlix/src/main/java/co/com/prueba/filmflix/domain/dto/ticket/TicketResponse.java
package co.com.prueba.filmflix.domain.dto.ticket;

import co.com.prueba.filmflix.domain.dto.film.FilmResponse;
import co.com.prueba.filmflix.domain.dto.function.FunctionResponse;
import co.com.prueba.filmflix.domain.dto.user.UserResponse;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;

@AllArgsConstructor
@Builder
@Data
public class TicketResponse {
    private Long id;
    private UserResponse user;
    private FilmResponse film;
    private FunctionResponse function;
    private String paymentMethod;
    private Integer quantity;
    private double total;
    private LocalDateTime date;
}

File: FilmFlixFront/src/app/domain/models/ticket/TicketResponse.ts
import { PaymentMethod } from '../../enums/PaymentMethod';
import { FilmResponse } from '../film/FilmResponse';
import { FunctionResponse } from '../function/FunctionResponse';
import { UserResponse } from '../user/UserResponse';

export interface TicketResponse {
    id: number;
    user: UserResponse;
    film: FilmResponse;
    function: FunctionResponse;
    paymentMethod: PaymentMethod;
    quantity: number;
    total: number;
    date: Date;
}
